compiler/rustc_target/src/asm/mips.rs

```diff
@@ -32,11 +32,13 @@ impl MipsInlineAsmRegClass {
 
     pub fn supported_types(
         self,
-        _arch: InlineAsmArch,
+        arch: InlineAsmArch,
     ) -> &'static [(InlineAsmType, Option<&'static str>)] {
-        match self {
-            Self::reg => types! { _: I8, I16, I32, F32; },
-            Self::freg => types! { _: F32; },
+        match (self, arch) {
+            (Self::reg, InlineAsmArch::Mips64) => types! { _: I8, I16, I32, I64, F32, F64; },
+            (Self::reg, _) => types! { _: I8, I16, I32, F32; },
+            (Self::freg, InlineAsmArch::Mips64) => types! { _: F32, F64; },
+            (Self::freg, _) => types! { _: F32; },
```

Comment:
Does MIPS32 not support `f64`? That sounds quite strange...

Reply:
With a simple test it seems that llvm support f64 on mips32.
But let's me check the mips reference for more details.
Since something have changed in mips32r6: some parts are deprecated.

Reply:
mips32 support f64 regs and their instructions but not i64.
I will add it then.